Tests/SourceKitLSPTests/LocalSwiftTests.swift

```diff
@@ -650,6 +650,7 @@ final class LocalSwiftTests: XCTestCase {
           code: nil,
           source: "sourcekitd",
           message: "initialization of immutable value \'a\' was never used; consider replacing with assignment to \'_\' or removing it",
+          tags: [.unnecessary],
```

Comment:
This is going to make the test fail using the currently released swift.  While we can't always guarantee sourcekit-lsp main branch will work with previously released toolchains, we do make an effort to keep it working when we can.

Since this particular test isn't about the diagnostic itself, I suggest changing this to

```swift
    // Diagnostic returned by code actions cannot be recursive.
    var expectedDiagnostic = diagnostic!
    expectedDiagnostic.codeActions = nil

    // Expected Fix-it: Replace `let a` with `_` because it's never used
    let expectedTextEdit = TextEdit(range: Position(line: 1, utf16index: 2)..<Position(line: 1, utf16index: 7), newText: "_")
    XCTAssertEqual(fixit, CodeAction(
      title: "Replace 'let a' with '_'",
      kind: .quickFix,
      diagnostics: [expectedDiagnostic],
...
```

Unfortunately this means we don't have a good way to test this new DiagnosticTag functionality, since it depends on the new sourcekitd.  For a more complex change we would probably need to teach the test to detect whether sourcekitd has the new feature or not and skip the test if it's not available.  In this particular case, detecting the presence of the new sourcekitd functionality in the test is almost as much work as the change itself here.  Since the sourcekit-lsp side of this change is relatively simple I'm okay with us not testing it for now.

Reply:
Okay, I have applied the suggestion :+1: